src/Form/formatters.test.js

```diff
@@ -200,5 +200,35 @@ describe('prepareFormValuesForStorage()', () => {
             });
             expect(values.credentials).toEqual({ password: inputCredentials.password });
         });
+
+        it('rejects credentials, if there is no credentials data and credit cards is available, and partial_form_saves is disabled', () => {
+            const values = prepareFormValuesForStorage(
+                {
+                    // @ts-ignore
+                    credentials: {},
+                    // @ts-ignore
+                    identities: {},
+                    // @ts-ignore
+                    creditCards: { cardNumber: '1234567890123456', cardName: 'Dax McDax' },
+                },
+                false,
+            ); // partial_form_saves is disabled
+            expect(values.credentials).toBeUndefined();
```

Comment:
Is it worth checking that `identities` is also undefined and that `creditCards` has the expected value?

Also, I wonder if we really need the two tests for when `partial_form_saves` is true or false. It doesn't seem to matter, does it? 🤔